CMakeLists.txt

```diff
@@ -410,6 +410,7 @@ if(QUDA_DOWNLOAD_EIGEN)
     DOWNLOAD_ONLY
     YES)
   target_include_directories(Eigen SYSTEM INTERFACE ${Eigen_SOURCE_DIR})
+  install(DIRECTORY ${Eigen_SOURCE_DIR}/Eigen TYPE INCLUDE)
```

Comment:
I am fine with this for now but technically Eigen is only used privately in QUDA and a calling application should not need it (the usual ones going through the interface should be fine). But maybe our headers are not set up for this yet. 
I guess if lyncs uses Eigen itself it should not use QUDA to get the headers.

Reply:
Ok I see what you mean. But actually, I had to add it due to an include error with the latest develop branch. Namely you have `eigen_helper.h` that is included by `eigensolve_quda.h` then included by `invert_quda.h`. So the installed headers are dependent on Eigen and without installing it I cannot load them.

Now, I have tried to remove the inclusion of `eigen_helper.h` and the first (and probably only) error I get is the following:
```
include/eigensolve_quda.h:201:61: error: unknown type name 'MatrixXi'
void permuteVecs(std::vector<ColorSpinorField> &kSpace, MatrixXi &mat, int size);
```
So as far as I can tell, currently, `Eigen` is not used only privately.